rules/do-you-optimize-tinacms-project-for-clarity-performance-and-reliable-builds/rule.md

```diff
@@ -0,0 +1,109 @@
+---
+type: rule
+tips: ""
+title: Do you Optimize Your TinaCMS Project for Clarity, Performance, and
+  Reliable Builds
+seoDescription: Learn effective strategies to structure and optimize your
+  TinaCMS project for clarity, performance, and successful builds.
+uri: do-you-optimize-tinacms-project-for-clarity-performance-and-reliable-builds
+authors:
+  - title: ""
+created: 2024-08-28T16:15:00.000Z
+guid: 3c59f56f-34b0-47a7-902a-718eb340d2e5
+---
+Structuring and optimizing your TinaCMS project is essential to achieve clarity, enhance performance, and prevent build failures. Poorly optimized projects can lead to slow site performance, increased server load, and even failed builds due to excessive or inefficient data requests.
+
+Let’s explore how to structure your project effectively and apply best practices to boost performance both in runtime and during the build process.
+
+## 1. Structuring Your TinaCMS Architecture
+
+When working with large datasets or generating multiple subcomponents, following best practices is crucial to maintain performance and clarity.
+
+### Bad Practices:
+
+* **Making Individual Requests with Tina Client for Each Subcomponent:**
+
+  * This method can overwhelm the build process, particularly when dealing with a large number of subcomponents.
+* **Using Deeply Nested Schemas with Nested References:**
+
+  * Complex and deeply nested schemas increase the complexity of the project, making it harder to manage and more prone to build failures.
+  * They can also lead to inefficient data fetching, further slowing down both runtime and build processes.
+
+### Good Practices:
+
+* **Making a Single Request at a Top-Level Server Component and Passing Props Down:**
+
+  * Data fetched at the top level is cached by default, which enhances performance by reducing redundant API calls.
+  * Subcomponents can directly access the necessary data, eliminating the need for repeated requests.
+
+![✅ Figure: Good example - Single request at the top-level server and passing props down](2024-08-28_16-21-56.png)
```

Comment:
```suggestion
::: good
![Figure: Good example - Single request at the top-level server and passing props down](2024-08-28_16-21-56.png)
::: 
```

Should use the directives/custom blocks instead of emojis

Reply:
✅ Done